### ISSUE TITLE
Update "Use my location" button/pattern

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#17118]
## Description 
On the Facility Locator we have an option for the user to "Use my location" when they are searching. There are some known issues with this currently:
- This is a `<button>` element however it does not visually look like a button. To maintain material honestly (elements should look like what they are) and to be more accessible and avoid confusion, if this is a `<button>`, it should look like a button
- ~There is no programmatic association between "Use my location" and the "City, State, Postal Code" field. This means that users may not know exactly what the purpose of this button is for, especially since this is before the input field - "Use my location for what?" There is [an existing ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12882) to add an `aria-describedby` attribute~
- When a user selects this button, but if they do not have location sharing turned on in their browser, a modal will pop up. The content in this modal isn't very robust, it would benefit from review and updating. Example - clearer instructions about how to enable location sharing?
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/4b57a457-b1eb-484b-81bc-9b7a3b017af1)

### Additional options
Amanda has pointed out that if we are considering updating this pattern, we don't necessarily need to stick to what we have exactly, there are other ways, out in the wild that this is done. [Zillow](https://www.zillow.com/) and [Redfin](https://www.redfin.com/) have "Current location" as a dropdown from the search input. This is definitely a way to do it and we could make it work, both of these have their quirks and aren't perfect as is.
- Zillow: Arrow keys are used to access "Current location" but there wasn't a clear SR announcement that this was available. Non-sighted uses may miss this as an available option
- Redfin: Tab is used to access "Current location" however it's not immediately after the search box, the X and search icon are inbetween. Also, when recent searches are available, the behavior becomes inconsistent and confusing because recent searches, while in the same dropdown are accesses with arrow keys and not tab.

## Broader implications
This "use my location" option is used in a few places throughout VA.gov but is not an established pattern or component. Because of that, the usage is is not standardized so the issues that were pointed out above might exist in other places and other instances might have potentially additional instances. There is an opportunity for us to improve this pattern, propose it as an experimental design to the DST and then other teams with "Use my location" would have an example of how to update and build their experience.

## Design

[Facility Locator Figma file](https://www.figma.com/file/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?type=design&node-id=1%3A17356&mode=design&t=qCpuSRmupMRlFrtW-1)

## Acceptance Criteria
- [x] Design is considered along with other Facility Locator updates in #17835 
- [x] Design is documented in Figma file
- [x] Design is reviewed with UX Lead, Product, Engineering, Content, and Accessibility
- [ ] Design is reviewed with and sent to VA UX Lead for final sign off
- [ ] Follow up tickets are created

**Comment by laflannery** [comment]
@aklausmeier In the FL Figma comments, you asked me to review the out in the wild implementations of "Use my location", this ticket includes that review as well as the overall issues CC: @mmiddaugh

**Comment by laflannery** [comment]
This is going to be discussed at the 2/9 DSC meeting apparently

**Comment by aklausmeier** [comment]
@mmiddaugh This is a ticket around FL "Use my location" we briefly discussed. If you happen to remember which prior research study might be related, let us know. 

I found some prior insights on Use my location in [this study](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/facilities/facility-locator/research/user-research/2021-mobile-experience-research/research-findings.md#2-most-participants-did-not-utilize-the-use-my-location-link). 

> 3 of 10 were hesitant to use the location feature on their phones in general
"I don’t usually use the, “use my location” option on most sites because I usually have the location of my phone turned off."

**Comment by aklausmeier** [comment]
for future self - another VA instance of Use my location button pattern on [Accredited Rep](https://www.va.gov/get-help-from-accredited-representative/find-rep) <img width="697" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOU41QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--637cd8bc2e3ab08cb3c76af5e5fb5e3c42c39c81/Screenshot%202024-04-19%20at%2010.06.47%20AM.png" alt="Screenshot 2024-04-19 at 10.06.47 AM.png" />

**Comment by aklausmeier** [comment]
Initial discussions on design iterations in [Figma](https://www.figma.com/file/GvNVeG9GAXZaBpny7vNxZP?type=design&node-id=12-10572&mode=design#682383830)

Might make sense to pull in with #17835

**Comment by jilladams** [comment]
@davidmpickett @thejordanwood For UX Sync tomorrow, we can refine and it might fit into end of Sprint 3.

**Comment by thejordanwood** [comment]
**Update from UX sync:**
Reviewed design with team and we're going to move forward with adding "Use my location" into a dropdown. Functionality should be:

- User selects location field
- "Use my location" should appear in dropdown
- If "Use my location" isn't selected, the dropdown will disappear when user starts typing into location field

<img width="844" alt="Screenshot 2024-08-29 at 3 25 23 PM" src="https://github.com/user-attachments/assets/5b27acd2-7678-4c6f-b0a7-994d56cc17d6">

**Next steps:**
@laflannery is going to determine how this should work best for accessibility, then we'll ask an engineer about tech feasibility. 

cc: @davidmpickett @aklausmeier

**Comment by thejordanwood** [comment]
Also noting that the ticket mentions changing the wording in the location modal. We proposed updated wording when we suggested new empty/error states for facility locator. This was the wording we proposed at the time, but let me know if it needs to be updated again. @davidmpickett @aklausmeier @laflannery 

<img width="334" alt="Screenshot 2024-08-29 at 3 47 26 PM" src="https://github.com/user-attachments/assets/d5dc8337-7625-4444-825e-9273d0da3b71">

**Comment by laflannery** [comment]
This still sounds good to me!

**Comment by thejordanwood** [comment]
### End of sprint update:

We now have 2 design options for this. Here's a [relevant Slack thread](https://dsva.slack.com/archives/C06S9DADEUB/p1725040518247339) that describes why Laura and I came up with a second option based on feedback she received in an a11y meeting.  The thread also has some discussion on next steps. 

**Option 1**
Adding "Use my location" into a dropdown.
<img width="864" alt="Screenshot 2024-09-03 at 4 28 52 PM" src="https://github.com/user-attachments/assets/13caa22d-7057-46a9-a59a-80b2425c3de7">

**Option 2**
Moving "Use my location" to a button on the right side of the location field, similar to how global search works on the site. 
<img width="864" alt="Screenshot 2024-09-03 at 4 30 00 PM" src="https://github.com/user-attachments/assets/b427f15f-1a47-4b42-89b2-e97f01b05fb7">

Laura, Eli, and I met to discuss both options and explore tech feasibility.
- Eli said the button option would be easier to implement
- We all agreed that the dropdown would be better IF we planned to implement a typeahead option to the location field sometime in the future (Example of this on [yelp.com](http://yelp.com/)). Then we could consider moving the 'Use my location' into a dropdown as a phase 1 approach.

**Next steps:**
I think at this point we need a product decision on which option to move forward with. Here are some open questions:

- Has a typeahead option for the facility locator field been considered in the past? 
- If not, could Eli look into if a typeahead option is feasible in the future? I think that would help us make a decision. 
- This may need usability testing (also referenced in [the Slack thread](https://dsva.slack.com/archives/C06S9DADEUB/p1725396381749289?thread_ts=1725040518.247339&cid=C06S9DADEUB)). Once we decide on a solution, we would need to think through next steps for that. 

cc: @jilladams @Agile6MSkinner @aklausmeier @laflannery @davidmpickett @eselkin

**Comment by jilladams** [comment]
Per Michelle: We don't know of previous work around typeahead on Location field, but recent convo around how "Springfield" could be MO, IL, etc. Mapbox would need to provide the validation, OR any other solution would need to be compatible with Mapbox.

@Agile6MSkinner Will make a SPIKE ticket to verify Mapbox typeahead, and that will finalize the  decision around Options 1 or 2. 

Closing, that in mind.
